cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -14,77 +14,100 @@ if(pyroot_experimental)
+    if(DEFINED Python2_VERSION AND "${Python2_VERSION}" VERSION_LESS "2.7")
+      message(WARNING "Ignoring Python2 installation: unsupported version ${Python2_VERSION} (version>=2.7 required)")
+    endif()
+    if(Python2_Development_FOUND AND "${Python2_VERSION}" VERSION_GREATER_EQUAL "2.7")
+      if(NOT Python3_Development_FOUND)
+        # Only Python2 was found, set as main
+        set(PYTHON_EXECUTABLE "${Python2_EXECUTABLE}" CACHE INTERNAL "" FORCE)
+        set(PYTHON_INCLUDE_DIRS "${Python2_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+        set(PYTHON_LIBRARIES "${Python2_LIBRARIES}" CACHE INTERNAL "" FORCE)
+        set(PYTHON_VERSION_STRING "${Python2_VERSION}" CACHE INTERNAL "" FORCE)
+        set(PYTHON_VERSION_MAJOR "${Python2_VERSION_MAJOR}" CACHE INTERNAL "" FORCE)
+        set(PYTHON_VERSION_MINOR "${Python2_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+        set(PYTHON_UNDER_VERSION_STRING "${Python2_VERSION_MAJOR}_${Python2_VERSION_MINOR}")
+        set(NUMPY_FOUND ${Python2_NumPy_FOUND} CACHE INTERNAL "" FORCE)
+        set(NUMPY_INCLUDE_DIRS "${Python2_NumPy_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+      else()
+        # Both Python3 and 2 found, set 2 as 'other'
+        set(OTHER_PYTHON_EXECUTABLE "${Python2_EXECUTABLE}")
+        set(OTHER_PYTHON_INCLUDE_DIRS "${Python2_INCLUDE_DIRS}")
+        set(OTHER_PYTHON_LIBRARIES "${Python2_LIBRARIES}")
+        set(OTHER_PYTHON_VERSION_STRING "${Python2_VERSION}")
+        set(OTHER_PYTHON_VERSION_MAJOR "${Python2_VERSION_MAJOR}")
+        set(OTHER_PYTHON_VERSION_MINOR "${Python2_VERSION_MINOR}")
+        set(OTHER_PYTHON_UNDER_VERSION_STRING "${Python2_VERSION_MAJOR}_${Python2_VERSION_MINOR}")
+        set(OTHER_NUMPY_FOUND ${Python2_NumPy_FOUND})
+        set(OTHER_NUMPY_INCLUDE_DIRS "${Python2_NumPy_INCLUDE_DIRS}")
+      endif()
     endif()
-  endif()
 
-  if(NOT Python3_Development_FOUND AND (NOT Python2_Development_FOUND OR "${Python2_VERSION}" VERSION_LESS "2.7"))
-    message(WARNING "No supported Python 2 or 3 development packages were found; PyROOT will not be built.")
-  endif()
+    if(NOT Python3_Development_FOUND AND (NOT Python2_Development_FOUND OR "${Python2_VERSION}" VERSION_LESS "2.7"))
+      message(WARNING "No supported Python 2 or 3 development packages were found; PyROOT will not be built.")
+    endif()
 
-  # Print message saying with which versions of Python are used to build
-  if(pyroot_experimental)
-    if(NOT Python3_Development_FOUND OR NOT Python2_Development_FOUND)
-      message(STATUS "Main Python used to build: ${PYTHON_VERSION_MAJOR}; PyROOT built for version ${PYTHON_VERSION_MAJOR} ")
-    elseif(Python3_Development_FOUND AND Python2_Development_FOUND)
-      message(STATUS "Main Python used to build: ${PYTHON_VERSION_MAJOR}; PyROOT built for versions ${PYTHON_VERSION_MAJOR} and ${OTHER_PYTHON_VERSION_MAJOR}")
+    # Print message saying with which versions of Python are used to build
+    if(pyroot_experimental)
```

Comment:
Why do we need to check only here for `pyroot_experimental`? Does all the code above work with old PyROOT?

Reply:
yes because the search for python is independent on pyroot/pyroot_experimental. The info messages protected by the if, on the other hand, are meant to be printer only for experimental

Reply:
So if I am correct, for the old PyROOT, we still do the search above for the two Python versions, but only one of them will be built anyway, even if the two of them are found. And here we only say for which Python version we are going to build, for the case of PyROOT experimental?

Reply:
exactly

Reply:
We don't have options enabled here still, it will not work.

Reply:
you need to move it somewhere else....

Reply:
@oshadura you mean we can't check here for `pyroot_experimental`? If so, we have another check for `pyroot_experimental` at the beginning of this file that we would need to solve too.

Reply:
No we can't check any options, this CMake file is "option-agnostic".